src/qibolab/_core/instruments/qm/program/arguments.py

```diff
@@ -27,6 +27,7 @@ class Parameters:
 
     element: Optional[str] = None
     lo_frequency: Optional[int] = None
+    max_offset: float = 0.5
```

Comment:
I have the feeling we start doing something wrong in here: `max_offset` seems not to be a property of `Parameters`, but just propagating a property of the instrument's configuration (or better, the port, since you should be able to control them individually).

Maybe it's fine to package it together with the other parameters, but I'm not fully convinced.
In general, this class seems to have a lot of parameters, many of which are only needed "sometimes", and others which are propagating parameters of something else.

Maybe we should rethink how sweepers are generated, and what they get access to.
In any case, this is definitely not urgent, and just a refactoring (if needed at all). So, it could be split as an issue. Unless there is a motivation to keep it as it is, in which case we could even save the issue.

Reply:
> Unless there is a motivation to keep it as it is, in which case we could even save the issue.

I would say there is no strong motivation. This initially started as a simple way to pass arguments required for the different sweep functions, mainly to simplify and unify them. However maybe it exploded a bit while adding features and different sweeper types (particularly baking, the non-interpolated duration sweep, now the different `max_offset`s, etc.). The only advantage I see is that the "mess" is concentrated in a single dataclass, but I am open to better suggestions. I would certainly not implement here and I wouldn't really consider it much of priority, but an issue should be okay.

> In general, this class seems to have a lot of parameters, many of which are only needed "sometimes", and others which are propagating parameters of something else.

The first part (up to ..."sometimes") is absolutely correct. I don't really get the second part (...parameters of something else), though.

Reply:
> but an issue should be okay.

Ok, I'll open it. And let's postpone the rest of the discussion.

>> , and others which are propagating parameters of something else.
> I don't really get the second part (...parameters of something else), though.

The `lo_frequency` should be already available as an attribute of the LO config, similarly the `max_offset` should be set as an attribute of a port/FEM configuration (or something like), and so on.
Instead of collecting all of them together, we may check whether we can refactor even the code downstream, to pass them separately, but even to functions with a smaller amount of arguments, and down a shorter chain of calls.
But it is a general consideration, I don't have any specific plan for this code. I should take more time to analyze it in depth (and you can do as well as myself - it's just that sometimes you need to take a step back, and reconsider things that grew adiabatically).

Reply:
Now in #1070

@stavros11 feel free to resolve if you are fine the second part.